app/views/portal/portal/home.html.erb

```diff
@@ -3,97 +3,105 @@
 <% content_for :card do %>
   <h1 class="h2"><%= t(".title", name: current_client.intake.preferred_name) %></h1>
 
-  <% if @current_step.present? %>
-    <p><%= t(".heres_what_we_need") %></p>
+  <% if @submit_additional_documents %>
     <div class="tax-return-status">
       <div class="tax-return-status__row tax-return-status__with-alert">
         <%= image_tag("icons/exclamation.svg", alt: "") %>
-        <% link_text = @current_step.include?("/documents/") ? t(".tax_returns.submit_additional_documents") : t(".tax_returns.answer_questions") %>
-        <%= link_to link_text, @current_step %>
+        <%= link_to t(".tax_returns.submit_additional_documents"), @current_step %>
       </div>
     </div>
-  <% else %>
+  <% end %>
 
-    <% @tax_returns.each do |tax_return| %>
-      <div class="tax-return-status" id=<%= "tax-year-#{tax_return.year}" %>>
-        <h2 class="h3"><%= t(".tax_returns.heading", year: tax_return.year) %></h2>
+  <% if @answer_questions %>
+    <div class="tax-return-status">
+      <div class="tax-return-status__row tax-return-status__with-alert">
+        <%= image_tag("icons/exclamation.svg", alt: "") %>
+        <%= link_to t(".tax_returns.answer_questions"), @current_step %>
+      </div>
+    </div>
+  <% end %>
+
+  <% if @tax_return_messages.present? %>
+    <% @tax_return_messages.each do |year, messages| %>
+      <div class="tax-return-status" id=<%= "tax-year-#{year}" %>>
+        <h2 class="h3"><%= t(".tax_returns.heading", year: year) %></h2>
         <div class="tax-return-status__content">
-          <!--
-            If there is a single final tax document for the tax year, label it with a fully-translatable generic title.
-            If there is more than one final tax document, label it with the provided document name to distinguish
-          !-->
-          <% if tax_return.final_tax_documents.present? %>
-            <% if tax_return.final_tax_documents.count == 1 %>
-              <div class="tax-return-status__row tax-return-status__solo">
-                <%= image_tag("icons/check.svg", alt: "") %>
-                <%= link_to t(".tax_returns.view_final_tax_document", year: tax_return.year), portal_document_path(id: tax_return.final_tax_documents.first.id), target: "_blank" %>
-              </div>
-            <% else %>
-              <% tax_return.final_tax_documents.each do |tax_doc| %>
-                <div class="tax-return-status__row tax-return-status__solo">
-                  <%= image_tag("icons/check.svg", alt: "") %>
-                  <%= link_to t(".tax_returns.view_document", display_name: tax_doc.display_name), portal_document_path(id: tax_doc.id), target: "_blank" %>
-                </div>
-              <% end %>
-            <% end %>
-          <% end %>
+          <% messages.each do |message| %>
+            <% tax_return = current_client.tax_returns.where(year: year).first %>
```

Comment:
I think this would be easier to read if you start by looping across the tax_returns - that way you can avoid the `<% tax_return = current_client.tax_returns.where(year: year).first %>`.

In that case, you might want to do something like:

`<% if @tax_return_messages[tax_return.year].include?(:final_tax_document) %>`

to get the same effect as the loop over messages